util/dynamicConfig.js

```diff
@@ -1,24 +1,38 @@
+/* eslint-disable no-param-reassign */
 const get = require("get-value");
 const unset = require("unset-value");
 
+function recurse(event, value, Config, field) {
+  value = value.replace("{{", "");
+  value = value.replace("}}", "");
+  if (value.includes("||")) {
+    const path = value.split("||")[0].trim();
+    const getFieldVal = get(event, path);
+    if (getFieldVal) {
+      Config[field] = getFieldVal;
+      unset(event, path);
+    } else {
+      Config[field] = JSON.parse(value.split("||")[1].trim()).toString();
+    }
+  }
+}
+
 function getDynamicConfig(event) {
   const { Config } = event.destination;
   if (Config) {
     Object.keys(Config).forEach(field => {
```

Comment:
Just a thought
How about write a new method with same signature using `recursion`, which makes things easier to read

Reply:
Sorry didn't get this. What do we want to include in new method?

Reply:
Leave it, it seems unnecessary now